org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java

```diff
@@ -332,6 +335,22 @@ private <T> T convertToCollectionType(InternalConverter converter) {
 		return convertToCollection(converter);
 	}
 
+	@SuppressWarnings("unchecked")
+	private <T> T convertToOptionalType(InternalConverter converter) {
+
+		if (object == null) {
```

Comment:
We should always attempt the conversion to the target type. There may be a custom handler which turns null into something else.

Reply:
check